src/wallet/walletutil.cpp

```diff
@@ -75,3 +75,28 @@ std::vector<fs::path> ListWalletDir()
 
     return paths;
 }
+
+WalletLocation::WalletLocation(const std::string& filename) : m_filename(filename)
+{
+    const fs::path wallet_dir = GetWalletDir();
+    m_path = fs::absolute(filename, wallet_dir);
+    fs::path name, parent = m_path;
+    while (!parent.empty()) {
+        if (parent == wallet_dir) {
+            break;
+        }
+        name = parent.filename() / name;
+        parent = parent.parent_path();
+    }
+    m_name = name.string();
+}
+
+std::string WalletLocation::GetDisplayName() const
+{
+    return m_name.empty() ? "[default wallet]" : m_name;
```

Comment:
This no longer puts brackets around the filename if it isn't the default wallet?